packages/api-page-builder-aco/src/page/hooks/onPageAfterCreate.hook.ts

```diff
@@ -0,0 +1,25 @@
+import WebinyError from "@webiny/error";
+
+import { createPageRecordPayload } from "~/utils/createRecordPayload";
+
+import { PbAcoContext, PbPageRecordData } from "~/types";
+
+export const onPageAfterCreateHook = (context: PbAcoContext) => {
+    const { aco, pageBuilder } = context;
+
+    /**
+     * Intercept page creation and create a new search record.
+     * The `location.folderId` info is coming from transient `meta` data.
+     */
+    pageBuilder.onPageAfterCreate.subscribe(async ({ page, meta }) => {
+        try {
+            const payload = await createPageRecordPayload(context, page, meta);
+            await aco.search.create<PbPageRecordData>(payload);
+        } catch (error) {
+            throw WebinyError.from(error, {
+                message: "Error while executing onPageAfterCreateHook hook",
+                code: "ACO_AFTER_PAGE_CREATE_HOOK"
+            });
```

Comment:
My question here is: do we really want to throw an error and kill the whole page creation because a search record didn't get stored? 🤔 Not sure if this is the right approach. @adrians5j any thoughts?

Reply:
In case of the search record creation error, the admin user won't be to list it in the page builder list. This is why I decided to kill the page creation.

Reply:
Ah, in that case it makes sense. I'm resolving this conversation.